### PR TITLE
Solution to assignment 4

### PR DESCRIPTION
**Pull request by EmaPajic**

### CODE REVIEW

File: EmaPajic/assignment4/assignment4.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+@author: EmaPajic
+"""
+
+def count_islands(rows,columns,tiles):
+    """
+    I am assuming that we can change tiles (set some values to false).
```
Comment (ariava) [inline EmaPajic/assignment4/assignment4.py:9]: Good job documenting your assumptions!

File: EmaPajic/assignment4/assignment4.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    If we couldn't do that, we could just make another matrix where we would store flags so that we don't visit same tile twice.
```
Comment (ariava) [inline EmaPajic/assignment4/assignment4.py]: Nit: please stay within 80 columns

File: EmaPajic/assignment4/assignment4.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    for i in range(0,rows):
+        for j in range(0,columns):
+            if tiles[i][j] == True:
+                numOfIslands += 1
+                find_all_parts_of_island(rows,columns,i,j,tiles)
+    return numOfIslands
+
+def valid_index(rows,columns,i,j):
+    # check if index is out of range
+    if i < 0 or i >= rows or j < 0 or j >= columns:
```
Comment (ariava) [inline EmaPajic/assignment4/assignment4.py]: You can also write something like
```
return not (i < 0 or i >= rows or j < 0 or j >= columns)
```
(or transform all conditions to positive)

File: EmaPajic/assignment4/test_assignment4.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import unittest
+from assignment4 import count_islands
+
+class Test(unittest.TestCase):
+    
+    def setUp(self):
+        self.tiles_empty = []
+        self.tiles1 = [[False,False],[False,False]]
+        self.tiles2 = [[True,True]]
+        self.tiles3 = [[True,True,True,True,True],[True,False,False,False,True],[True,False,True,False,True],[True,False,False,False,True],[True,True,True,True,True]]
```
Comment (ariava) [inline EmaPajic/assignment4/test_assignment4.py]: Please put the different sub-arrays each on a different line and use spacing, so it's easier to understand where are the islands in your example.
E.g.
```
[[True , True , True , True , True],
 [True , False, False, False, True],
 [True , False, True , False, True],
 [True , False, False, False, True],
 [True , True , True , True , True]]
```

**Review by ariava — CHANGES_REQUESTED** [review]
Super good job! You were so fast! Please take a look at my comments.

File: EmaPajic/assignment4/assignment4.py (hunk 2 above)
Comment (EmaPajic) [reply EmaPajic/assignment4/assignment4.py]: updated :)

File: EmaPajic/assignment4/assignment4.py (hunk 3 above)
Comment (EmaPajic) [reply EmaPajic/assignment4/assignment4.py]: updated :)

File: EmaPajic/assignment4/test_assignment4.py (hunk 4 above)
Comment (EmaPajic) [reply EmaPajic/assignment4/test_assignment4.py]: updated :)

**Review by ariava — APPROVED** [review]
Awesome!

**Pull request merged**